crates/matrix-sdk/tests/integration/client.rs

```diff
@@ -898,3 +902,238 @@ async fn create_dm_error() {
 
     assert_eq!(client_api_error.status_code, 404);
 }
+
+#[async_test]
+async fn ambiguity_changes() {
+    let (client, server) = logged_in_client().await;
+
+    let example_id = user_id!("@example:localhost");
+    let example_2_id = user_id!("@example2:localhost");
+    let example_3_id = user_id!("@example3:localhost");
+
+    // Initial sync, adds 2 members.
+    mock_sync(&server, &*test_json::SYNC, None).await;
+    let response = client.sync_once(SyncSettings::default()).await.unwrap();
+    server.reset().await;
```

Comment:
Why is this `reset` necessary? If it's for an obscure reason, can you please add a comment about it?

Reply:
Either we need to pass the sync token manually in `mock_sync`, or we need to reset the server between syncs so the response matches the latest call to `mock_sync` without the sync token.

Tests are already split between both methods.